src/Streamly/Internal/Data/Fold/Type.hs

```diff
@@ -1119,16 +1221,21 @@ chunksOf2 n (Fold step1 initial1 extract1) (Fold2 step2 inject2 extract2) =
 -- XXX We can use asyncClock here. A parser can be used to return an input that
 -- arrives after the timeout.
 -- XXX If n is 0 return immediately in initial.
+-- XXX we should probably discard the input received after the timeout like
```

Comment:
Maybe keep it unreleased till we fix the behaviour?

Reply:
We can do that.